src/details/ArborX_Point.hpp

```diff
@@ -23,42 +23,47 @@ class Point
 private:
   struct Data
   {
-    double coords[3];
+    float coords[3];
   } _data = {};
 
+  struct Abomination
+  {
+    double xyz[3];
+  };
+
 public:
   KOKKOS_INLINE_FUNCTION
   constexpr Point() noexcept = default;
 
   KOKKOS_INLINE_FUNCTION
-  constexpr Point(Data data)
-      : _data{std::move(data)}
+  constexpr Point(Abomination data)
+      : Point(data.xyz[0], data.xyz[1], data.xyz[2])
```

Comment:
Make sure you see that.  This allow the narrowing in aggregate initialization.  It is not pretty but that way I don't have to go change all occurrences yet.  I would prefer to defer fixing this to a later PR.